src/widgets/canvas/lv_canvas.c

```diff
@@ -386,9 +390,17 @@ void lv_canvas_init_layer(lv_obj_t * obj, lv_layer_t * layer)
 void lv_canvas_finish_layer(lv_obj_t * canvas, lv_layer_t * layer)
 {
     if(layer->draw_task_head == NULL) return;
+
+    bool task_dispatched;
+
     while(layer->draw_task_head) {
         lv_draw_dispatch_wait_for_request();
-        lv_draw_dispatch_layer(lv_obj_get_display(canvas), layer);
+        task_dispatched = lv_draw_dispatch_layer(lv_obj_get_display(canvas), layer);
+
+        if(!task_dispatched) {
+            lv_draw_wait_for_finish();
+            lv_draw_dispatch_request();
```

Comment:
Is this allowed to be called outside of a draw driver like this?

Reply:
What do you mean?

Reply:
Yes it's valid. When `lv_canvas_finish_layer` is called drawing is running and it's okay to handle dispatching here.